Springer Link.js

```diff
@@ -225,8 +226,23 @@ function complementItem(doc, item) {
 
 	let docType = ZU.xpathText(doc, '//meta[@name="citation_article_type"]/@content');
 	if (docType.match(/(Book R|reviews?)|(Review P|paper)/)) item.tags.push("Book Reviews");
+	// ORCID
+	getORCID(doc, item);
 	return item;
 }
+	// ORCID
+function getORCID(doc, item) {
+	let authorOrcidEntries = ZU.xpath(doc, '//*[@class="c-article-author-list__item"]');
+	for (let authorOrcidEntry of authorOrcidEntries) {
+		let authorEntry = authorOrcidEntry.innerText;
+		let orcidEntry = authorOrcidEntry.innerHTML;
+		if (authorEntry && orcidEntry.match(/\d+-\d+-\d+-\d+x?/i)) {
```

Comment:
Please test `orcidEntry` for undefined before executing `match`.

Reply:
Do you mean like `if (authorEntry && typeof orcidEntry !== 'undefined' && orcidEntry.match(/\d+-\d+-\d+-\d+x?/i))` or should be enough like this? `if (authorEntry && orcidEntry && orcidEntry.match(/\d+-\d+-\d+-\d+x?/i))`

Reply:
The second one should be sufficient.